packages/firebase_messaging/example/test_driver/firebase_messaging.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:io';
+import 'dart:async';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+
+void main() {
+  final Completer<String> completer = Completer<String>();
+  enableFlutterDriverExtension(handler: (_) => completer.future);
+  tearDownAll(() => completer.complete(null));
+
+  group('$FirebaseMessaging', () {
+    final FirebaseMessaging firebaseMessaging = FirebaseMessaging();
+
+    test('getToken', () async {
+      final String token = await firebaseMessaging.getToken();
+      // Token is sometimes null in simulators
+      // expect(token, isNotNull);
+    });
+
+    test('autoInitEnabled', () async {
+      await firebaseMessaging.setAutoInitEnabled(false);
+      expect(await firebaseMessaging.autoInitEnabled(), false);
+      await firebaseMessaging.setAutoInitEnabled(true);
+      expect(await firebaseMessaging.autoInitEnabled(), true);
+    });
+
+    test('subscribeToTopic', () async {
+      firebaseMessaging.subscribeToTopic('foo');
```

Comment:
Some idea, doesn't need to be in this PR. Maybe we can make `subscribeToTopic` and `unsubscribeFromTopic` to return a Future<bool> to indicate the invocation to the platform is happened correctly. That'd make the integration test easier.

Reply:
I agree, and will update in a subsequent PR